packages/muelu/src/MueCentral/MueLu_VerbosityLevel.hpp

```diff
@@ -109,6 +110,40 @@ namespace MueLu {
   //!
   VerbLevel toMueLuVerbLevel(const Teuchos::EVerbosityLevel verbLevel);
 
+  //! global variable that holds the default output rank for new VerboseObjects
+  extern int defaultMueLuOutputRank;
+
+  //! constant that is interpreted as "output on default rank"
+  extern const int OUTPUT_TO_ROOT;
+
+  //! On all MPI ranks that are contained in oldMap, set the default output rank for MueLu to the lowest available rank that is present in newMap.
+  //! This is used in rebalancing, since the original root output might have been dropped.
+  template <class LocalOrdinal, class GlobalOrdinal, class Node>
+  void updateDefaultMueLuOutputRank(Teuchos::RCP<const typename Xpetra::Map<LocalOrdinal, GlobalOrdinal, Node> > oldMap,
+                                    Teuchos::RCP<const typename Xpetra::Map<LocalOrdinal, GlobalOrdinal, Node> > newMap) {
+#ifdef HAVE_MPI
```

Comment:
1. I don't see why you need to check `HAVE_MPI` or use MPI's interface here; the algorithm should work fine with SerialComm.

2. Don't use `HAVE_MPI`; use the namespaced macro `HAVE_TEUCHOS_MPI`.